### ISSUE TITLE
ExponentialSmoothing Model Produces Different Results on Mac vs Linux

### ISSUE THREAD

**Issue by justinessert** [statsmodels/statsmodels#8894]
## Description

I am running an `ExponentialSmoothing` model and am seeing differences in the output on Linux vs Mac. I'm trying my best to reconcile these differences to see if I can get them to produce the same results regardless of OS.

## Versions Used

I didn't see anything indicating that these results were sensitive to much to the versions being ran, but for the sake of consistency, I fixed the following versions between environments:

- Python: 3.8.16
- Numpy: 1.24.3
- Pandas: 1.5.3
- Statsmodels: 0.14.0
- Scipy: 1.10.1

## Code to Reproduce

```python
from statsmodels.tsa.holtwinters import ExponentialSmoothing
import pandas as pd

df = pd.read_csv("~/Downloads/AirPassengers.csv")

X = df["#Passengers"].values
ts = df["Month"].values

model = ExponentialSmoothing(
    X,
    trend=None,
    damped_trend=False,
    seasonal=None,
    seasonal_periods=None,
    use_boxcox=False,
    dates=ts,
).fit(
    smoothing_level=None,
    smoothing_trend=None,
    smoothing_seasonal=None,
    use_brute=False,
)

result = model.predict(start=ts[0], end=ts[-1])
print(result.mean())
```

> The dataset being used here can be found on Kaggle [here](https://www.kaggle.com/datasets/rakannimer/air-passengers?resource=download)

This code gives me the following results:

- On my Mac: `275.84320334735634`
- On Linux:     `278.07638822684953`

Further, If I instead check `model.params` I see that the differences arise before even predicting:

On my Mac:

```python
{'smoothing_level': 0.5,
 'smoothing_trend': nan,
 'smoothing_seasonal': nan,
 'damping_trend': nan,
 'initial_level': 118.46666666666658,
 'initial_trend': nan,
 'initial_seasons': array([], dtype=float64),
 'use_boxcox': False,
 'lamda': None,
 'remove_bias': False}
```

On Linux:

```python
{'smoothing_level': 0.9999999850981204,
 'smoothing_trend': nan,
 'smoothing_seasonal': nan,
 'damping_trend': nan,
 'initial_level': 111.99990880905746,
 'initial_trend': nan,
 'initial_seasons': array([], dtype=float64),
 'use_boxcox': False,
 'lamda': None,
 'remove_bias': False}
```

Would someone be able to help me understand why I am getting these differences?

**Comment by justinessert** [comment]
## Smoothers.holt__

### initial_level's calculation

Based on this analysis, it seems like the `smoothers.holt__` seems to be the issue. Based on my understanding, this is the the function, defined by `statsmodels` which is used in `scipy`'s  `minimize` as the function to optimize. See:

- Defining `smoothers` dict [here](https://github.com/statsmodels/statsmodels/blob/77cb066320391ffed4196a32491ddca28e8c9122/statsmodels/tsa/holtwinters/model.py#L63)
- Defining `func` [here](https://github.com/statsmodels/statsmodels/blob/77cb066320391ffed4196a32491ddca28e8c9122/statsmodels/tsa/holtwinters/model.py#L816)
- Wrapped and passed into Scipy's `minimize` [here](https://github.com/statsmodels/statsmodels/blob/77cb066320391ffed4196a32491ddca28e8c9122/statsmodels/tsa/holtwinters/model.py#L898)

If I run the following code I still get different results:

```python
import pandas as pd

df = pd.read_csv("../data/air_passengers.csv")

X = df["#Passengers"].values
ts = df["Month"].values

import numpy as np
from statsmodels.tsa.holtwinters._exponential_smoothers import HoltWintersArgs
from statsmodels.tsa.holtwinters import _exponential_smoothers as smoothers

hwargs_sel =  np.array([1, 0, 0, 1, 0, 0])
params = np.array([
    0.5, np.nan, np.nan, 118.466666666667, np.nan, 1.,
])
bounds = np.array([[0., 1.,],
 [0., 1.,],
 [0., 1.,],])
y = X.astype(float)
m = 0
nobs = 144
hw_args = HoltWintersArgs(
    hwargs_sel, params, bounds, y, m, nobs
)

x = np.array([[0.5, 118.466666666667]]).flatten()
z = smoothers.holt__(x, hw_args)
z.T @ z
```

Results:

- On my Mac: `249044.8982723232`
- On Linux:     `249044.89827232316`

It seems like there is a slight difference here that could be causing the differences in the output of ExponentialSmoothing, because even a small difference in the output of the function being optimized could potentially cause wide differences in the overall results.

### smoothing_level calculation

I was also able to see a similar affect when analyzing the calculation of `smoothing_level` (for this analysis, I fixed the `initial_level` to avoid the first issue). I followed down this trace:

- (source) `ExponentialSmoothing.fit`
- (statsmodels) [ExponentialSmoothing._optimize_parameters](https://github.com/statsmodels/statsmodels/blob/77cb066320391ffed4196a32491ddca28e8c9122/statsmodels/tsa/holtwinters/model.py#L1143)
- (statsmodels) [minimize](https://github.com/statsmodels/statsmodels/blob/77cb066320391ffed4196a32491ddca28e8c9122/statsmodels/tsa/holtwinters/model.py#L900)
- (scipy) [_minimize_slsqp](https://github.com/scipy/scipy/blob/19d7a9461e24ce110077f9b075ec745fd48bb037/scipy/optimize/_minimize.py#L719)
- (scipy) [wrapped_grad](https://github.com/scipy/scipy/blob/6adfc144e63ca18a501b2f9b96991fe59f1db755/scipy/optimize/_slsqp_py.py#L416)
- (scipy) [sf.grad](https://github.com/scipy/scipy/blob/6adfc144e63ca18a501b2f9b96991fe59f1db755/scipy/optimize/_slsqp_py.py#L380). The `sf` variable is initialized via:
    - (scipy) [_prepare_scalar_function](https://github.com/scipy/scipy/blob/6adfc144e63ca18a501b2f9b96991fe59f1db755/scipy/optimize/_slsqp_py.py#L374)
    - (scipy)  [ScalarFunction](https://github.com/scipy/scipy/blob/6adfc144e63ca18a501b2f9b96991fe59f1db755/scipy/optimize/_optimize.py#L382)
- (scipy) [self._update_grad](https://github.com/scipy/scipy/blob/6adfc144e63ca18a501b2f9b96991fe59f1db755/scipy/optimize/_differentiable_functions.py#L273)
- (scipy) [self._update_grad_impl](https://github.com/scipy/scipy/blob/6adfc144e63ca18a501b2f9b96991fe59f1db755/scipy/optimize/_differentiable_functions.py#L256)
- (scipy) [update_grad](https://github.com/scipy/scipy/blob/6adfc144e63ca18a501b2f9b96991fe59f1db755/scipy/optimize/_differentiable_functions.py#L176)
- (scipy) [approx_derivative](https://github.com/scipy/scipy/blob/6adfc144e63ca18a501b2f9b96991fe59f1db755/scipy/optimize/_differentiable_functions.py#L173)
- (scipy) [_dense_difference](https://github.com/scipy/scipy/blob/6adfc144e63ca18a501b2f9b96991fe59f1db755/scipy/optimize/_numdiff.py#L505)
- (scipy) [fun](https://github.com/scipy/scipy/blob/6adfc144e63ca18a501b2f9b96991fe59f1db755/scipy/optimize/_numdiff.py#L576)

I was able to see similar results, this time if I ran

```python
import pandas as pd

df = pd.read_csv("../data/air_passengers.csv")

X = df["#Passengers"].values
ts = df["Month"].values

import numpy as np
from statsmodels.tsa.holtwinters.model import opt_wrapper
from statsmodels.tsa.holtwinters import (
    _exponential_smoothers as smoothers
)

val = 118.467

## HW args
hwargs_sel =  np.array([1, 0, 0, 1, 0, 0])
params = np.array([
    0.5, np.nan, np.nan, val, np.nan, 1.,
])
bounds = np.array([[0., 1.,],
 [0., 1.,],
 [0., 1.,],])
y = X.astype(float)
m = 0
nobs = 144
hw_args = smoothers.HoltWintersArgs(
    hwargs_sel, params, bounds, y, m, nobs
)
hw_args.xi = hwargs_sel
hw_args.transform = True

func = opt_wrapper(smoothers.holt__)
def func_wrapped(x):
    return func(x, hw_args)

xx = np.array([0.5000000149011612, 118.467])
func_wrapped(xx)
```

Then I get still get differing results:

- On my Mac: `fun(x): 249044.89479089883`
- On Linux:     `fun(x): 249044.8947908988`

### TLDR

Based on from what I can tell, the difference is originating from the `smoothers.holt__` function providing an extra significant digit sometimes. But in the first example the extra sig digit appeared on Linux, whereas in the second example, it appeared on my Mac... not sure how this could flip-flop?

**Comment by bashtage** [comment]
The optimizer on the mac is doing a bad job.  I think it is possible the default should change.  Add `method="L-BFGS-B"` to the `fit` call and I think you will see the same.

**Comment by justinessert** [comment]
I have yet to confirm this, but I'd imagine that the issue I described above:

> Based on from what I can tell, the difference is originating from the smoothers.holt__ function providing an extra significant digit sometimes. But in the first example the extra sig digit appeared on Linux, whereas in the second example, it appeared on my Mac... not sure how this could flip-flop?

Would still affect the optimization, right? Even if there was a different optimization method used?
